### PR TITLE
Fix missing diagnostics for nodes

### PR DESCRIPTION
**Pull request by MadHed**
Fixes a bug where nodes that are inside a children array were not checked for diagnostics.
Also adds support for different parser severity levels.

### CODE REVIEW

File: src/TreeAnalyzer.php [hunk 1]
```diff
@@ -52,8 +52,34 @@ public function __construct(PhpParser\Parser $parser, string $content, DocBlockF
...
+    private function collectDiagnostics($node)
+    {
+        if (($error = PhpParser\DiagnosticsProvider::checkDiagnostics($node)) !== null) {
+            $range = PhpParser\PositionUtilities::getRangeFromPosition($error->start, $error->length, $this->sourceFileNode->fileContents);
+
+            if ($error->kind == \Microsoft\PhpParser\DiagnosticKind::Error) {
+                $severity = DiagnosticSeverity::ERROR;
+            } else {
+                $severity = DiagnosticSeverity::WARNING;
+            }
```
Comment (felixfbecker) [inline src/TreeAnalyzer.php]: could you use a `switch` for this? that would it clearer that you are doing a 1:1 mapping

File: src/TreeAnalyzer.php [hunk 2]
```diff
@@ -52,8 +52,34 @@ public function __construct(PhpParser\Parser $parser, string $content, DocBlockF
         $this->collectDefinitionsAndReferences($this->sourceFileNode);
     }
 
+    private function collectDiagnostics($node)
```
Comment (felixfbecker) [inline src/TreeAnalyzer.php]: add a docblock

File: src/TreeAnalyzer.php (hunk 1 above)
Comment (MadHed) [reply src/TreeAnalyzer.php]: Sure

**Comment by MadHed** [comment]
TBH, I would actually like to refactor out the mutual recursion between `collectDefinitionsAndReferences` and `update`. What are your thoughts on that?

**Comment by felixfbecker** [comment]
If the resulting code is clearer, sure.

**Pull request merged**